libs/header/p2p/exchange_test.go

```diff
@@ -382,6 +382,38 @@ func TestExchange_RequestHeadersFromAnotherPeerWhenTimeout(t *testing.T) {
 	assert.NotEqual(t, newPeerScore, prevScore)
 }
 
+// TestExchange_RequestPartialRange enusres in case of receiving a partial response
+// from server, Exchange will re-request remaining headers from another peer
+func TestExchange_RequestPartialRange(t *testing.T) {
+	hosts := createMocknet(t, 3)
+	exchg, _ := createP2PExAndServer(t, hosts[0], hosts[1])
+
+	// create one more server(with more headers in the store)
+	serverSideEx, err := NewExchangeServer[*test.DummyHeader](
+		hosts[2], headerMock.NewStore[*test.DummyHeader](t, test.NewTestSuite(t), 10),
+		WithNetworkID[ServerParameters](networkID),
+	)
+	require.NoError(t, err)
+	require.NoError(t, serverSideEx.Start(context.TODO()))
+	exchg.peerTracker.peerLk.Lock()
+	prevScoreBefore1 := exchg.peerTracker.trackedPeers[hosts[1].ID()].peerScore
+	prevScoreBefore2 := 50
+	// reducing peerScore of the second server, so our exchange will request host[1] first.
+	exchg.peerTracker.trackedPeers[hosts[2].ID()] = &peerStat{peerID: hosts[2].ID(), peerScore: 50}
+	exchg.peerTracker.peerLk.Unlock()
+	h, err := exchg.GetRangeByHeight(context.Background(), 1, 8)
+	require.NotNil(t, h)
+	require.NoError(t, err)
+
+	exchg.peerTracker.peerLk.Lock()
+	prevScoreAfter1 := exchg.peerTracker.trackedPeers[hosts[1].ID()].peerScore
+	prevScoreAfter2 := exchg.peerTracker.trackedPeers[hosts[2].ID()].peerScore
+	exchg.peerTracker.peerLk.Unlock()
```

Comment:
The fact that we need to modify peertracker tin each test hints us that something is wrong with the API. Not a concern of this PR, but it is good to keep that in mind and try to avoid that

Reply:
Why? peerScore is an internal logic of the peerTracker and has to be hidden(encapsulated) without any chance to get the write access to it. All peerScore updates should be done in a single way(updateStats). From the other side, we are trying to cover as edges many cases as we can using UT, so for these purposes, I do not see anything bad in manual modifying.

Reply:
No, it's fine. Just did not see anything bad in this, but only in cases when logic is hidden and only for unit tests. Also, I'm not sure that we can unify these API to be able to cover all our needs in UT. There still could be tricky cases...